torch/ao/quantization/experimental/quantizer.py

```diff
@@ -1,13 +1,64 @@
 import torch
 from torch import Tensor
+from torch.ao.quantization.experimental.observer import APoTObserver
+from torch.ao.quantization.experimental.apot_utils import float_to_apot, float_to_reduced_precision
 
 # class to store APoT quantizer
 # implements quantize and dequantize
 # and stores all quantization parameters
 class APoTQuantizer(torch.Tensor):
-    @staticmethod
-    def quantize_APoT(tensor2quantize: Tensor) -> Tensor:
-        raise NotImplementedError
+    b: int
+    k: int
+    n: int
+    signed: bool
+    use_int_repr: bool
+    quantization_levels: torch.Tensor
+    level_indices: torch.Tensor
+    data: torch.Tensor
+
+    def __init__(
+        self,
+        b,
+        k,
+        signed,
+            dtype=torch.quint8) -> None:
+        self.signed = signed
+
+        # check for valid inputs of b, k
+        assert(k and k != 0)
+        assert(b % k == 0)
+        self.b = b
+        self.k = k
+        self.n = b // k
+
+        # make observer, get quantizion levels and level indices
+        obs = APoTObserver(max_val=1.0, b=b, k=k)
+        obs_result = obs.calculate_qparams(signed=signed)
+        self.quantization_levels = obs_result[1]
+        self.level_indices = obs_result[2]
+
+    r""" Quantizes fp Tensor to integer or reduced precision fp representation, depending on user input.
+    Conversion is based on the calculated quantization levels from a specified APoT non-uniform observer.
+    The approach follows the method outlined in the APoT paper: https://arxiv.org/pdf/1909.13144.pdf.
+    Args:
+        tensor2quantize: fp Tensor
+        use_int_repr: bool flag to specify int of reduced precision fp representation of APoT tensor
```

Comment:
what is the difference between True and False? can you give an example?

Reply:
looks like reduced_precision is the dequantized fp value. I feel we should only be storing int values in `data` so that representation is in integer

Reply:
@jerryzh168 I think reduced_precision is the quantized levels in fp representation (powers of 2) so I think it's analogous to dequantized fp values in uniform quantzation?

Reply:
yeah, but we don't really store dequantized fp values in uniform quantization, the reason we quantize things is so that we can store things in int8 I think. I'm wondering why do we store this?

Reply:
i'll modify so that we don't store reduced precision representation anymore (just int8 representation)